src/user.rs

```diff
@@ -1,148 +1,127 @@
-				display_name: self.get_display_name(),
-			}]
-		} else {
-			vec![]
-		}
+	/// Get a display name for this user
+	pub fn get_display_name(&self) -> String {
+		format!("{}, {}", self.last_name, self.first_name)
 	}
-}
 
-impl From<ZitadelUser> for ImportHumanUserRequest {
-	fn from(user: ZitadelUser) -> Self {
-		Self {
-			user_name: user.user_data.email.clone().to_string(),
-			profile: Some(Profile {
-				first_name: user.user_data.first_name.clone().to_string(),
-				last_name: user.user_data.last_name.clone().to_string(),
-				display_name: user.get_display_name(),
-				gender: Gender::Unspecified.into(), // 0 means "unspecified",
-				nick_name: user.user_data.external_user_id.clone().to_string(),
-				preferred_language: String::default(),
-			}),
-			email: Some(Email {
-				email: user.user_data.email.clone().to_string(),
-				is_email_verified: !user.needs_email_verification,
-			}),
-			phone: user.user_data.phone.as_ref().map(|phone| Phone {
-				phone: phone.to_owned().to_string(),
-				is_phone_verified: !user.needs_phone_verification,
-			}),
-			password: String::default(),
-			hashed_password: None,
-			password_change_required: false,
-			request_passwordless_registration: false,
-			otp_code: String::default(),
-			idps: user.get_idps(),
-		}
+	/// Get the external user ID in raw byte form
+	pub fn get_external_id_bytes(&self) -> Result<Vec<u8>> {
+		// This looks ugly at a glance, since we get the original
+		// bytes at some point, however some users will be retrieved
+		// from Zitadel at a later point, so we cannot assume that we
+		// know the original bytes, and must always decode the
+		// external user ID to get those.
+		hex::decode(&self.external_user_id).context("Invalid external user ID")
 	}
-}
 
-impl Display for ZitadelUser {
-	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-		write!(f, "email={}", &self.user_data.email)
+	/// Get the famedly UUID of this user
+	pub fn get_famedly_uuid(&self) -> Result<String> {
+		Ok(Uuid::new_v5(&FAMEDLY_NAMESPACE, self.get_external_id_bytes()?.as_slice()).to_string())
 	}
-}
 
-/// A structure that can either be a string or bytes
-#[derive(Clone, Debug)]
-pub(crate) enum StringOrBytes {
-	/// A string
-	String(String),
-	/// A byte string
-	Bytes(Vec<u8>),
+	/// Get a base64-encoded external user ID, if the ID is raw bytes,
+	/// or a UTF-8 string if not.
+	///
+	/// Note: This encoding scheme is inherently broken, because it is
+	/// impossible to tell apart base64 encoded strings from
+	/// non-base64 encoded strings. We can therefore never know if the
+	/// ID should be decoded or not when re-parsing it, and it may
+	/// create collisions (although this is unlikely).
+	///
+	/// Only use this for Zitadel support.
+	pub fn get_string_id(&self) -> Result<String> {
+		let id = self.get_external_id_bytes()?;
+		Ok(if let Ok(encoded_id) = String::from_utf8(id.clone()) {
+			encoded_id
+		} else {
+			BASE64_STANDARD.encode(id)
+		})
+	}
```

Comment:
This seems to be forgotten for the HEX encoding update.

Reply:
No, give the doc comment another read. Zitadel uses base64 encoding to map binary IDP IDs, and will not understand our hex-encoded IDs. We need to encode IDs as base64 *specifically* for the cases in which zitadel actually reads them.

Maybe we should move this to `zitadel.rs` though?

Reply:
Got it. I will move it to `zitadel.rs`